module/VuFind/src/VuFind/ILS/Driver/PAIA.php

```diff
@@ -1774,6 +1786,48 @@ protected function paiaPostAsArray($file, $data)
      */
     protected function paiaLogin($username, $password)
     {
+        // as PAIA supports two authentication methods (defined as grant_type:
+        // password or client_credentials), check which one is configured
+        if (!in_array($this->grantType, ['password', 'client_credentials'])) {
+            throw new ILSException(
+                'Unsupported PAIA grant_type configured: ' . $this->grantType
+            );
+        }
+
+        // prepare http header
+        $header_data = [];
+
+        // prepare post data depending on configured grant type
+        switch ($this->grantType) {
+        case 'password':
+            $post_data = [
+                "username"   => $username,
+                "password"   => $password
+            ];
+            break;
+        case 'client_credentials':
+            // client_credentials only works if we have client_credentials
+            // username and password
+            if (isset($this->config['PAIA']['client_username'])
+                && isset($this->config['PAIA']['client_password'])
+            ) {
+                $header_data["Authorization"] = 'Basic ' .
+                    base64_encode(
+                        $this->config['PAIA']['client_username'] . ':' .
+                        $this->config['PAIA']['client_password']
+                    );
+                $post_data = [
```

Comment:
As I mentioned in the above comment, I do not understand, why we need global config settings at this point. Shouldn't we rather use the username and password the user provided in the login form (stored in $username and $password)?